python/tank/context.py

```diff
@@ -899,22 +901,40 @@ def serialize(context):
         "additional_entities": context.additional_entities,
         "_pc_path": context.tank.pipeline_configuration.get_path()
     }
+
+    from tank_vendor import shotgun_authentication as sg_auth
+
+    # If there is a current user.
+    user = get_current_user()
+    if user:
+        # We should serialize it as well.
+        data["current_user"] = sg_auth.serialize_user(user)
     return pickle.dumps(data)
-    
-    
+
+
 def deserialize(context_str):
     """
-    Deserializaes a string created with serialize() into a context object
+    Deserializes a string created with serialize() into a context object
     """
     # lazy load this to avoid cyclic dependencies
-    from .api import Tank
+    from .api import Tank, set_current_user
+    from tank_vendor import shotgun_authentication as sg_auth
     
     data = pickle.loads(context_str)
 
     # first get the pc path out of the dict
     pipeline_config_path = data["_pc_path"] 
     del data["_pc_path"]
-    
+
+    # See if there is a current user set.
```

Comment:
I think this is a really important part of the workflow around authentication. The current comment describes what the next two lines of code are doing - It would be great if there was an explanation of the overall flow here too - how this allows an authenticated user to travel between processes, DCC launches etc. And how this current_user concept connects to the general authentication!